src/huggingface_hub/hub_mixin.py

```diff
@@ -212,20 +212,39 @@ def __init_subclass__(
         # Will be reused when creating modelcard
         tags = tags or []
         tags.append("model_hub_mixin")
-        cls._hub_mixin_info = MixinInfo(
-            model_card_template=model_card_template,
-            repo_url=repo_url,
-            docs_url=docs_url,
-            model_card_data=ModelCardData(
-                languages=languages,
-                library_name=library_name,
-                license=license,
-                license_name=license_name,
-                license_link=license_link,
-                pipeline_tag=pipeline_tag,
-                tags=tags,
-            ),
-        )
+
+        # Initialize MixinInfo if not existent
+        if getattr(cls, "_hub_mixin_info", None) is None:
+            cls._hub_mixin_info = MixinInfo(
+                model_card_template=model_card_template,
+                model_card_data=ModelCardData(),
+            )
+        info = cls._hub_mixin_info
+
+        # Update MixinInfo with metadata
+        if model_card_template is not None and model_card_template != DEFAULT_MODEL_CARD:
+            info.model_card_template = model_card_template
+        if repo_url is not None:
+            info.repo_url = repo_url
+        if docs_url is not None:
+            info.docs_url = docs_url
+        if languages is not None:
+            info.model_card_data.language = languages
+        if library_name is not None:
+            info.model_card_data.library_name = library_name
+        if license is not None:
+            info.model_card_data.license = license
+        if license_name is not None:
+            info.model_card_data.license_name = license_name
+        if license_link is not None:
+            info.model_card_data.license_link = license_link
+        if pipeline_tag is not None:
+            info.model_card_data.pipeline_tag = pipeline_tag
+        if tags is not None:
+            if info.model_card_data.tags is not None:
+                info.model_card_data.tags.extend(tags)
+            else:
+                info.model_card_data.tags = tags
```

Comment:
Should we extend tags instead of overriding them? In that case, it would be impossible to drop base model tags and set new ones. Probably there is no best solution, but I would expect them to be replaced rather than extended 🙂 WDYT?

```suggestion
            info.model_card_data.tags = tags
```

Reply:
I would prefer to extend them actually, specifically so that we don't loose tags defined by base models. Worst case, users can overwrite the internal attribute :confused: 
TBH I don't think this will ever be a problem, multiple inheritance when using ModelHubMixin is already not common so cases where nested classes define tags and where these tags should not be inherited might just never happen.

Reply:
Make sense, thank you!